pkg/insights/insightsclient/insightsclient.go

```diff
@@ -61,19 +60,19 @@ type Source struct {
 var ErrWaitingForVersion = fmt.Errorf("waiting for the cluster version to be loaded")
 
 // New creates a Client
-func New(client *http.Client, maxBytes int64, metricsName string, authorizer Authorizer, clusterInfo ClusterVersionInfo) *Client {
+func New(client *http.Client, maxBytes int64, metricsName string, authorizer Authorizer, gatherKubeConfig *rest.Config) *Client {
```

Comment:
Just out of curiosity - why is it called `gatherKubeConfig` ? Wouldn't just `kubeConfig` be suficient?

Reply:
True, but the reason I named it `gatherKubeConfig` is that it uses the `gatherKubeConfig` from the `operator.go` which is the specific config for gathering the data that the `insightsclient` will need to gather, so I think its clearer this way (for me atleast :D ).

The same could be brought up in the gatherer, it also uses `gatherKubeConfig`, plus `gatherProtoKubeConfig` and `metricsGatherKubeConfig`. (`gatherKubeConfig` and `gatherProtoKubeConfig` differ very little)

So we have several configs and I would like to keep note which part should use which and I know no better way then this.

Reply:
Ahh right. Thanks for explanation. Makes sense then....and I should probably check why we have more of these.